src/orchard/xref.clj

```diff
@@ -13,21 +32,87 @@
     (symbol? thing) (var-get (find-var thing))
     (fn? thing) thing))
 
-(defn fn-deps
-  "Returns a set with all the functions invoked by `val`.
-  `val` can be a function value, a var or a symbol."
-  {:added "0.5"}
-  [val]
-  (let [val (as-val val)]
-    (set (some->> val class .getDeclaredFields
+(defn- f->sym [^clojure.lang.AFn f]
+  (-> f .getClass .getName repl/demunge symbol))
+
+(defn- fn-source [f]
+  (hunt-down-source (f->sym f)))
+
+(defn- fn-name [^java.lang.Class f]
+  (-> f .getName repl/demunge symbol))
+
+(defonce classbytes (atom {}))
+
+(defn- recompile [ns-sym form]
+  (push-thread-bindings
+   {clojure.lang.Compiler/LOADER
+    (proxy [clojure.lang.DynamicClassLoader] [@clojure.lang.Compiler/LOADER]
```

Comment:
Just as a word of caution, hacking with classloaders as a long history of giving us problems, in various tools over the years.

If you have the chance, please avoid these as they can compose badly with other features.

Clojure internals are best kept internal :)